### DISCUSSION TITLE
Provide an amalgamated header somewhere

### DISCUSSION THREAD

**Discussion by parkertomatoes** [taocpp/PEGTL#188] [General]
It would be great if the amalgamated header was checked in somewhere or made available from an official URL. 

That might sound like a petty request, but that would make it usable from [Compiler Explorer](http://godbolt.org), which would in turn make it easier to test and share snippets of code using PEGTL.

For an example of this with a different library, here is a snippet using the amalgamated header from the nlohmann json parsing library:
https://godbolt.org/z/2-dmgE

That can be shared on Stack Overflow or Medium, or used as a scratchpad for testing ideas.

**Comment by d-frey** [comment]
The PEGTL can already be chosen as a supported library on Compiler Explorer, you don't need an amalgamated header for that. Click on "Libraries" and select whether you want "2.8.0" or "trunk". Then simply include PEGTL headers.

Example: https://godbolt.org/z/Q2hZnz

This allows to PEGTL to be used with different systems/settings, an amalgamated header will always reflect the system itself in some way (as different internal headers and `#ifdef`-sections are included depending on which compiler and which compiler version is used).

**Comment by parkertomatoes** [comment]
Ha, I've been doing it wrong all along! Thanks for the informative response.
